src/neo4j/neo4j-document.js

```diff
@@ -29,12 +37,33 @@ export async function addDocumentToNeo4j(doc) {
   let arrEdges = [];
   let docElements = doc.elements();
   for (const e of docElements) {
-    if (e.isEntity()) {
+    if (e.isEntity() && !e.isComplex()) {
       let nodeInfo = {
         id: `${e.association().dbPrefix}:${e.association().id}`,
         name: e.association().name
       };
       arrNodes.push(nodeInfo);
+    } else if (e.isEntity && e.isComplex()) {
+      // TODO: Push on the edges that will represent this complex e in arrEdges
+      // Untested
+      const complex = e;
+      const component = makeComponent(complex, doc);
+      for (let i = 0; i < complex.elements.length - 2; i++) {
```

Comment:
Is this creating the in-complex edges? If so will these indices work?

Reply:
Yes- so complex has a field of entries. Suppose `entries: [A, B, C, D]`. 

We take the first entry, `i=0 entries[ i ] = A`. We make an edge between `A` and entry that follows it in `entries` (this is what the inner loop/the`j` index is for). So `A` is connected to `B`, `C`, `D`.
After that is done, increment `i, i=1` and do the same for `B`. B is connected to `C`, `D` (and from the previous iteration, `A` is also connected to `B`).

By the time we get to D, the last element of entries, we know that `A`, `B` and `C` are already connected to `D` so there is nothing to do.

I could also just let the outer loop's conditional be `complex.elements().length - 1` to make things simpler. The inner loop's conditional won't be satisfied anyway. 

What do you think?

Reply:
I have altered `line 51`'s condition to be `complex.elements().length - 1`

Reply:
Note to self: the `complex.elements().length - 2` condition would not work (because of case where there are 2 nodes- both loops would never run in that case). 
It must be `complex.elements().length - 1`

Thank you for catching that error @jvwong!